apps/studio/src/lib/db/authentication/azure.ts

```diff
@@ -153,6 +171,31 @@ export class AzureAuthService {
     }
   }
 
+  private async getAzureCLIToken(): Promise<AuthConfig | null> {
+    return new Promise((resolve, reject) => {
+      const command = "az account get-access-token --resource https://ossrdbms-aad.database.windows.net --output json";
+      exec(command, { encoding: "utf8" }, (error, stdout) => {
+        if (error) {
+          console.error("Error getting token:", error);
+          reject(null);
+          return;
+        }
+        try {
+          const tokenData = JSON.parse(stdout);
+          resolve(  {
+            type: 'azure-active-directory-default',
+            options: {
+              token: tokenData.accessToken,
+            }
+          });
+        } catch (parseError) {
+          console.error("Error parsing token JSON:", parseError);
+          reject(null);
```

Comment:
same question about the reject. How does the upstream code know what error happened?